internal/lm/nvml.go

```diff
@@ -218,6 +233,90 @@ func newGPUModeLabeler(devices []resource.Device) (Labeler, error) {
 	return labels, nil
 }
 
+func newImexDomainLabeler(configFile string, device []resource.Device) (Labeler, error) {
+	if configFile == "" {
+		return nil, nil
+	}
+
+	// Read file and parse it
+	imexConfig, err := os.Open(configFile)
+	if err != nil && os.IsNotExist(err) {
+		return nil, nil
+	}
+	if err != nil {
+		return nil, fmt.Errorf("failed to open imex config file: %v", err)
+	}
+	defer imexConfig.Close()
+
+	// Read the file line by line
+	var ips []string
+	scanner := bufio.NewScanner(imexConfig)
+	for scanner.Scan() {
+		ip := strings.TrimSpace(scanner.Text())
+		if net.ParseIP(ip) == nil {
+			return nil, fmt.Errorf("invalid IP address in imex config file: %s", ip)
+		}
+		ips = append(ips, ip)
+	}
+
+	if err := scanner.Err(); err != nil {
+		return nil, fmt.Errorf("failed to read imex config file: %v", err)
+	}
+
+	if len(ips) == 0 {
+		return nil, nil
+	}
+
+	// Sort the IP addresses
+	sort.Strings(ips)
+
+	// Join the sorted IPs into a single string
+	sortedIPs := strings.Join(ips, "\n")
```

Comment:
It may be more readable to generate the config ID here instead of where we consume `sortedIPs`. We're not really interested in the IPs, but the ID associated with the unique set.

Reply:
Agreed. I in fact, i would reorganize this whole function to get the clusterUUID and cliqueIDs first, and then generate the IMEX domain ID and then inserting it directly in the labels without this intermediate veriable.